contract-message-test/tests/harness.rs

```diff
@@ -1,184 +1,672 @@
-
-    // Get provider and client
-    let provider = wallet.get_provider().unwrap();
-    let client = &provider.client;
-
-    // Get padded bytecode root that must be hardcoded into the predicate to constrain the spending transaction
-    let mut script_bytecode =
-        std::fs::read("../contract-message-script/out/debug/contract_message_script.bin")
-            .unwrap()
-            .to_vec();
-    let padding = script_bytecode.len() % 8;
-    let script_bytecode_unpadded = script_bytecode.clone();
-    script_bytecode.append(&mut vec![0; padding]);
-    let script_hash = Hasher::hash(&script_bytecode);
-
-    println!("Padded script length: {}", script_bytecode.len());
-    println!("Padded script hash   : 0x{:?}", script_hash);
-
-    // Deploy test contract
-    let test_contract_id = Contract::deploy(
-        "../contract-message-test/out/debug/contract_message_test.bin",
-        &wallet,
-        TxParameters::default(),
-        StorageConfiguration::default(),
-    )
-    .await
-    .unwrap();
-    let contract_instance = TestContract::new(test_contract_id.to_string(), wallet.clone());
-    println!("Test contract_id   : 0x{:?}", test_contract_id);
-
-    // Get predicate bytecode and root
-    let predicate_bytecode =
-        std::fs::read("../contract-message-predicate/out/debug/contract_message_predicate.bin")
+pub const RANDOM_SALT: &str = "0x1a896ebd5f55c10bc830755278e6d2b9278b4177b8bca400d3e7710eee293786";
+pub const RANDOM_SALT2: &str = "0xd5f55c10bc830755278e6d2b9278b4177b8bca401a896eb0d3e7710eee293786";
+
+// Test that input messages can be relayed to a contract
+// and that the contract can successfully parse the message data
+mod success {
+    use std::str::FromStr;
+
+    use crate::utils::environment as env;
+    use fuels::prelude::Contract;
+    use fuels::prelude::Salt;
+    use fuels::prelude::StorageConfiguration;
+    use fuels::prelude::TxParameters;
+    use fuels::test_helpers::DEFAULT_COIN_AMOUNT;
+    use fuels::tx::{
+        Address, AssetId, Bytes32, ContractId, Input, Output, TxPointer, UtxoId, Word,
+    };
+
+    pub const RANDOM_SALT: &str =
+        "0x1a896ebd5f55c10bc830755278e6d2b9278b4177b8bca400d3e7710eee293786";
+    pub const RANDOM_SALT2: &str =
+        "0xd5f55c10bc830755278e6d2b9278b4177b8bca401a896eb0d3e7710eee293786";
+
+    #[tokio::test]
+    async fn relay_message_with_predicate_and_script_constraint() {
+        let data_word = 54321u64;
+        let data_bytes = Bytes32::from_str(RANDOM_SALT).unwrap();
+        let data_address = Address::from_str(RANDOM_SALT2).unwrap();
+        let mut message_data = data_word.to_be_bytes().to_vec();
+        message_data.append(&mut env::decode_hex(RANDOM_SALT));
+        message_data.append(&mut env::decode_hex(RANDOM_SALT2));
+        let message_data = env::prefix_contract_id(message_data).await;
+        let message = (100, message_data);
+        let coin = (DEFAULT_COIN_AMOUNT, AssetId::default());
+
+        // Set up the environment
```

Comment:
When you name something well `setup_environment` comments like `// Set up the environment` become redundant. Comments should be used to provide additional insight that might not be obvious. A repetition of a statement does not add further value and rather can be regarded towards ... for a lack of a better word, spam.

As an example of what is a superior comment `// Verify test contract received the message` provides insight into some message being sent and received and thus being able to extract the value, in this case `1`.

Reply:
Yeah, I think you're right especially in this case when it's a comment for one line of code. Some of the other comments are more there as a separator, but I guess that's what whitespace is for, lol.